azurerm/resource_arm_virtual_network_peering.go

```diff
@@ -147,17 +147,18 @@ func resourceArmVirtualNetworkPeeringRead(d *schema.ResourceData, meta interface
 		return fmt.Errorf("Error making Read request on Azure virtual network peering %q: %+v", name, err)
 	}
 
-	peer := *resp.VirtualNetworkPeeringPropertiesFormat
-
 	// update appropriate values
 	d.Set("resource_group_name", resGroup)
 	d.Set("name", resp.Name)
 	d.Set("virtual_network_name", vnetName)
-	d.Set("allow_virtual_network_access", peer.AllowVirtualNetworkAccess)
-	d.Set("allow_forwarded_traffic", peer.AllowForwardedTraffic)
-	d.Set("allow_gateway_transit", peer.AllowGatewayTransit)
-	d.Set("use_remote_gateways", peer.UseRemoteGateways)
-	d.Set("remote_virtual_network_id", peer.RemoteVirtualNetwork.ID)
+
+	if peer := resp.VirtualNetworkPeeringPropertiesFormat; peer != nil {
+		d.Set("allow_virtual_network_access", peer.AllowVirtualNetworkAccess)
+		d.Set("allow_forwarded_traffic", peer.AllowForwardedTraffic)
+		d.Set("allow_gateway_transit", peer.AllowGatewayTransit)
+		d.Set("use_remote_gateways", peer.UseRemoteGateways)
+		d.Set("remote_virtual_network_id", peer.RemoteVirtualNetwork.ID)
```

Comment:
there's another potential crash here; could we make this:

```suggestion
		if network := peer.RemoteVirtualNetwork; network != nil {
			d.Set("remote_virtual_network_id", .ID)
		}
```